common/rpc/grpc.go

```diff
@@ -145,3 +145,18 @@ func ServiceErrorInterceptor(
 	resp, err := handler(ctx, req)
 	return resp, serviceerror.ToStatus(err).Err()
 }
+
+func FrontendErrorInterceptor(
+	ctx context.Context,
+	req interface{},
+	_ *grpc.UnaryServerInfo,
+	handler grpc.UnaryHandler,
+) (interface{}, error) {
+	resp, err := handler(ctx, req)
+
+	// mask some internal errors at frontend
+	if _, ok := err.(*serviceerrors.ShardOwnershipLost); ok {
+		err = serviceerror.NewUnavailable("shard unavailable, please backoff and retry")
+	}
+	return resp, serviceerror.ToStatus(err).Err()
```

Comment:
instead of duplicating the code in `ServiceErrorInterceptor`, this could be an interceptor that sits just inside of it and only rewrite the error.